hotshot-query-service/src/data_source/update.rs

```diff
@@ -168,19 +171,45 @@ fn genesis_vid<Types: NodeType>(
 ) -> anyhow::Result<(VidCommonQueryData<Types>, VidShare)> {
     let payload = Payload::<Types>::empty().0;
     let bytes = payload.encode();
-    let mut disperse = advz_scheme(GENESIS_VID_NUM_STORAGE_NODES)
-        .disperse(bytes)
-        .context("unable to compute VID dispersal for genesis block")?;
-    ensure!(
-        VidCommitment::V0(disperse.commit) == leaf.block_header().payload_commitment(),
-        "computed VID commit {} for genesis block does not match header commit {}",
-        disperse.commit,
-        leaf.block_header().payload_commitment()
-    );
-    Ok((
-        VidCommonQueryData::new(leaf.block_header().clone(), Some(disperse.common)),
-        VidShare::V0(disperse.shares.remove(0)),
-    ))
+
+    match leaf.block_header().payload_commitment() {
+        VidCommitment::V0(commit) => {
+            let mut disperse = advz_scheme(GENESIS_VID_NUM_STORAGE_NODES)
+                .disperse(bytes)
+                .context("unable to compute VID dispersal for genesis block")?;
+
+            ensure!(
+                disperse.commit == commit,
+                "computed VID commit {} for genesis block does not match header commit {}",
+                disperse.commit,
+                commit
+            );
+            Ok((
+                VidCommonQueryData::new(leaf.block_header().clone(), Some(disperse.common)),
+                VidShare::V0(disperse.shares.remove(0)),
+            ))
+        }
+        VidCommitment::V1(commit) => {
+            let avidm_param = init_avidm_param(GENESIS_VID_NUM_STORAGE_NODES)?;
+            let weights = vec![1; GENESIS_VID_NUM_STORAGE_NODES];
+            let ns_table = parse_ns_table(bytes.len(), &leaf.block_header().metadata().encode());
+
+            let (calculated_commit, mut shares) =
+                AvidMScheme::ns_disperse(&avidm_param, &weights, &bytes, ns_table).unwrap();
+
+            ensure!(
+                calculated_commit == commit,
+                "computed VID commit {} for genesis block does not match header commit {}",
+                calculated_commit,
+                commit
+            );
+
+            Ok((
+                VidCommonQueryData::new(leaf.block_header().clone(), None),
```

Comment:
So the new VID scheme does not use VidCommon anymore? Should we make this a bit more clear by having an enum Type for V1 and V2 instead?

Reply:
Chengyu is going to add it